### PR TITLE
build: tweak CircleCI config

### PR DESCRIPTION
**Pull request by P0lip**
Related to https://github.com/stoplightio/spectral/issues/1835

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,6 @@
     "build": "tsc --build ./tsconfig.build.json",
     "postbuild": "yarn workspace @stoplight/spectral-cli postbuild",
     "lint": "yarn lint.prettier && yarn lint.eslint && yarn lint.changelog",
-    "lint.changelog": "kacl lint",
```
Comment (P0lip) [inline package.json:31]: not needed as of v6.x

File: .github/workflows/auto-approve.yml [hunk 2]
```diff
@@ -1,17 +0,0 @@
-name: Auto-Dependabot
```
Comment (P0lip) [inline .github/workflows/auto-approve.yml:1]: wasn't used + it's unsafe to update deps without any surveillance.

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+  windows: circleci/windows@2.4.1
 
 executors:
   linux:
     docker:
-      - image: cimg/base:2021.01
+      - image: cimg/base:2021.07
+
+defaults:
+  default-branch: &default-branch develop
```
Comment (P0lip) [inline .circleci/config.yml:12]: this can be change to whatever default, and will be used by all relevant tasks

**Review by marbemac — APPROVED** [review]
Didn't notice anything wrong, but I trust that you have tested all of this

**Pull request merged**